sched/adaptdl_sched/controller.py

```diff
@@ -135,11 +135,22 @@ async def _sync_job(self, namespace, job_name):
             elif allocation and not pods:
                 # Start the next group of pods.
                 job["status"]["group"] = job["status"].get("group", -1) + 1
-                for rank in range(len(allocation)):
-                    await self._create_pod(
-                        job["metadata"],
-                        job["spec"]["template"], allocation,
-                        job["status"]["group"], rank)
+                try:
+                    for rank in range(len(allocation)):
+                        await self._create_pod(
+                            job["metadata"],
+                            job["spec"]["template"], allocation,
+                            job["status"]["group"], rank)
+                except kubernetes.client.rest.ApiException as e:
+                    LOG.info(f"Failed to create pod for {job_name}. "
+                             "Setting job status to failed")
+                    job["status"]["phase"] = "Failed"
+                    job["status"]["reason"] = "PodCreationError"
+                    job["status"]["message"] = str(e)
+                    pods = await self._core_api.list_namespaced_pod(
+                        namespace, label_selector=f"adaptdl/job={job_name}")
```

Comment:
You can reuse the pods returned by `create_namespaced_pod` instead of querying the API server again. Also use a different variable name to avoid confusion with the `pods` defined on line 103.